fix(nemesis publisher): add Scylla commit date to ES

fgelcer opened the pull request:
up until now, we were saving only Scylla version
and the git sha of the commit, making chart creation less clear or logical, since we didn't have the
commit dates to order the latest entries.

From now, the entries will have commit date,
allowing us to order the chart by commit date,
and showing the last entries, being more
meaningful for chart presentation.

Refs: scylladb/qa-tasks#1489

## PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

fruch commented:
I would recommend writing a script to update the previous results (one year back)

fgelcer commented:
> I would recommend writing a script to update the previous results (one year back)

how would we know what is the commit date of the commit SHAs already saved?

fruch commented:
> > I would recommend writing a script to update the previous results (one year back)
> 
> how would we know what is the commit date of the commit SHAs already saved?

from scylla git repo

fgelcer commented:
> > > I would recommend writing a script to update the previous results (one year back)
> > 
> > 
> > how would we know what is the commit date of the commit SHAs already saved?
> 
> from scylla git repo

i will work on it, and after running it, will submit the script, in a new PR

roydahan commented:
1. This is not enough, in this case when I'm choosing a field in kibana I need to decide if to use version / date / commit.
I would like to have one field that has the full version pattern (version.date.commit).
OR as I asked in the original task change the commit to be date.commit.
I'm happy with both, but prefer the first one.

2. No need to update old results, we will manage with onward only.

fgelcer commented:
> 1. This is not enough, in this case when I'm choosing a field in kibana I need to decide if to use version / date / commit.
>    I would like to have one field that has the full version pattern (version.date.commit).
>    OR as I asked in the original task change the commit to be date.commit.
>    I'm happy with both, but prefer the first one.
> 2. No need to update old results, we will manage with onward only.

in addition to the separated cases (version, date, commitID), i added also the "full version", like you asked (version.date.commit)

and, because it is many times useful, i added also the build-id

fgelcer commented:
> > > I would recommend writing a script to update the previous results (one year back)
> > 
> > 
> > how would we know what is the commit date of the commit SHAs already saved?
> 
> from scylla git repo

i will try to work on that in the background, but for now, let's try to get it merged ASAP, so tonight's run will already include this new collected parameters

fruch commented on sdcm/nemesis_publisher.py:
```diff
@@ -49,21 +49,30 @@ def publish(self, disrupt_name, status=True, data=None):
             job_name=test_data['test_details']['job_name'],
             test_name=test_data['test_details']['test_name'],
-            scylla_version=version,
+            version=version,
```
I wouldn't recommend doing a switch like that 

Now all of the data won't be aligned, some item would show scylla version in on format and some in a different format.

And all of the dashboards would need to be fixed.

fgelcer replied:
fixed

fgelcer commented:
giving it a run [here](https://argus.scylladb.com/workspace?state=WyI3NjY4OGQ2ZC1kYzdjLTRmZWYtYmFmZC1kN2JlMzhkYzhhYTkiXQ)

fgelcer commented:
> giving it a run [here](https://argus.scylladb.com/workspace?state=WyI3NjY4OGQ2ZC1kYzdjLTRmZWYtYmFmZC1kN2JlMzhkYzhhYTkiXQ)

we have a result:
```
{
  "_index": "nemesis_data",
  "_type": "nemesis",
  "_id": "_1SkhYoBRaDOrL_jvE2Q",
  "_version": 1,
  "_score": 1,
  "_source": {
    "test_id": "a297a3cb-4ae8-41cb-b0f3-29b4fc4ec6bb",
    "job_name": "scylla-staging/fabio/nemesis-publisher/nemesis-decommission",
    "test_name": "longevity_test.LongevityTest.test_custom_time",
    "version": "5.4.0~dev",
    "scylla_git_sha": "0656810c2877",
    "scylla_commit_date": "20230910",
    "scylla_version": "5.4.0~dev.20230910.0656810c2877",
    "build_id": "2156440484bcf26ee80adb62c33ea5d7f689a57c",
    "nemesis_name": "NodetoolDecommission",
    "nemesis_duration": 1429,
    "start_time": "2023-09-11T18:44:10",
    "end_time": "2023-09-11T19:07:59",
    "target_node": "Node longevity-5gb-1h-DecommissionMonkey-db-node-a297a3cb-3 [34.247.66.105 | 10.4.1.47] (seed: True)",
    "outcome": "passed"
  },
  "fields": {
    "test_id.keyword": [
      "a297a3cb-4ae8-41cb-b0f3-29b4fc4ec6bb"
    ],
    "full_scylla_version.keyword": [
      "5.4.0~dev.20230910.0656810c2877"
    ],
    "scylla_git_sha.keyword": [
      "0656810c2877"
    ],
    "target_node": [
      "Node longevity-5gb-1h-DecommissionMonkey-db-node-a297a3cb-3 [34.247.66.105 | 10.4.1.47] (seed: True)"
    ],
    "scylla_git_sha": [
      "0656810c2877"
    ],
    "scylla_version.keyword": [
      "5.4.0~dev"
    ],
    "outcome": [
      "passed"
    ],
    "scylla_commit_date.keyword": [
      "20230910"
    ],
    "test_id": [
      "a297a3cb-4ae8-41cb-b0f3-29b4fc4ec6bb"
    ],
    "full_scylla_version": [
      "5.4.0~dev.20230910.0656810c2877"
    ],
    "nemesis_name.keyword": [
      "NodetoolDecommission"
    ],
    "build_id.keyword": [
      "2156440484bcf26ee80adb62c33ea5d7f689a57c"
    ],
    "nemesis_duration": [
      1429
    ],
    "outcome.keyword": [
      "passed"
    ],
    "end_time": [
      "2023-09-11T19:07:59.000Z"
    ],
    "nemesis_name": [
      "NodetoolDecommission"
    ],
    "scylla_version": [
      "5.4.0~dev"
    ],
    "start_time": [
      "2023-09-11T18:44:10.000Z"
    ],
    "target_node.keyword": [
      "Node longevity-5gb-1h-DecommissionMonkey-db-node-a297a3cb-3 [34.247.66.105 | 10.4.1.47] (seed: True)"
    ],
    "test_name.keyword": [
      "longevity_test.LongevityTest.test_custom_time"
    ],
    "job_name": [
      "scylla-staging/fabio/nemesis-publisher/nemesis-decommission"
    ],
    "job_name.keyword": [
      "scylla-staging/fabio/nemesis-publisher/nemesis-decommission"
    ],
    "build_id": [
      "2156440484bcf26ee80adb62c33ea5d7f689a57c"
    ],
    "scylla_commit_date": [
      "20230910"
    ],
    "test_name": [
      "longevity_test.LongevityTest.test_custom_time"
    ]
  }
}
```
